NitroxModel/Helper/NetHelper.cs

```diff
@@ -124,5 +137,31 @@ public static IPAddress GetHamachiIp()
             }
             return null;
         }
+
+        /// <summary>
+        ///     Returns true if the given IP address is reserved for private networks.
+        /// </summary>
+        public static bool IsPrivate(this IPAddress address)
+        {
+            static bool IsInRange(IPAddress ipAddress, string mask)
+            {
+                string[] parts = mask.Split('/');
+
+                int ipNum = BitConverter.ToInt32(ipAddress.GetAddressBytes(), 0);
+                int cidrAddress = BitConverter.ToInt32(IPAddress.Parse(parts[0]).GetAddressBytes(), 0);
+                int cidrMask = IPAddress.HostToNetworkOrder(-1 << (32 - int.Parse(parts[1])));
+
+                return (ipNum & cidrMask) == (cidrAddress & cidrMask);
+            }
+
```

Comment:
Maybe we can check if the adress.IsIPv4MappedToIpv6 to convert the adress to a standard IPV4 using MapToIpv4() method, will allow us to cover more case

Reply:
This seems too niche to be worth adding at this moment.